src/DepthImageToLaserScanROS.cpp

```diff
@@ -47,17 +47,20 @@
 namespace depthimage_to_laserscan{
 
 DepthImageToLaserScanROS::DepthImageToLaserScanROS(const rclcpp::NodeOptions & options): rclcpp::Node("depthimage_to_laserscan", options){
-  auto qos = rclcpp:: SystemDefaultsQoS();
-  cam_info_sub_ = this->create_subscription<sensor_msgs::msg::CameraInfo>("depth_camera_info", qos,
+  std::string depth_camera_info_topic = this->declare_parameter("depth_camera_info_topic", "depth_camera_info");
```

Comment:
There's no reason to provide a parameter for the topic to use; that's already enabled by the remapping capabilities of the ROS 2 core, something like:

```
ros2 run dethimage_to_laserscan depth_image_to_laserscan --ros-args -r depth_camera_info:=foo
```

So please remove this parameter (and the one below).  The change to QoS setting is a more interesting piece here, and I'll have to do more research/thinking to see if we want that.